agent/eventhandler/handler.go

```diff
@@ -28,6 +28,8 @@ var statesaver statemanager.Saver = statemanager.NewNoopStateManager()
 
 func HandleEngineEvents(taskEngine engine.TaskEngine, client api.ECSClient, saver statemanager.Saver) {
 	statesaver = saver
+	handler := NewTaskHandler()
```

Comment:
What's the division of labor between the two? It seems like we should have only one of these if we're looking to consolidate these in a single entity.